lambda_function.py

```diff
@@ -0,0 +1,629 @@
+import json
+import logging
+import sys
+import traceback
+import os
+import datetime
+import re
+
+import boto3
+
+conf = dict()
+conf['log_level'] = os.environ.get('LOG_LEVEL', 'INFO')
+conf['dynamodb_table_name'] = os.environ.get('DYNAMODB_TABLE_NAME', 'dns-manager')
+
+route53 = boto3.client('route53')
+dynamodb = boto3.client('dynamodb')
+
+# Apply logging configuration
+log = logging.getLogger(__name__)
+log.setLevel(conf['log_level'])
+FORMAT = '%(asctime)s %(levelname)s %(message)s'
+logging.basicConfig(format=FORMAT)
+
+
+def validate_event(event):
+    '''Check if request has valid input and permissions to manage records.'''
+    if isinstance(event['body'], str):
+        event['body'] = json.loads(event['body'])
+    print(json.dumps({'event': event}, default=str))
+    data = event['body']
+    if len(str(data)) > 512:
+        #log.error('Input is to long')
+        print(json.dumps({'message': 'Input is to long'}, default=str))
+        sys.exit('Input is to long')
+    try:
+        json_test = json.dumps(data)
+    except:
+        #log.error('json from event["body"] is not valid')
+        print(json.dumps({'message': 'json from event["body"] is not valid'}, default=str))
+        sys.exit('json from event["body"] is not valid')
+    validate_event_keys(data)
+    validate_event_action(data)
+    validate_event_username(data)
+    validate_event_nameservers(data)
+    validate_event_zoneid(data)
+    validate_event_domainname(data)
+    validate_event_email(data)
+
+def validate_event_action(data):
+    '''Check if request has valid Action key.'''
+    # Action unknown
+    if data.get('Action') not in ['CREATE', 'DELETE', 'UPDATE']:
+        #log.error('Action is not valid')
+        print(json.dumps({'message': 'Action is not valid'}, default=str))
+        sys.exit('Action is not valid')
+
+def validate_event_keys(data):
+    '''Check if request has mandatory keys.'''
+    # Not mandatory 'NameServers' handler for 'Action': 'DELETE'
+    if data['Action'] == 'DELETE' and 'NameServers' not in data:
+        data['NameServers'] = 'fake1,fake2'
+    for key in ['Action', 'UserName', 'NameServers', 'ZoneID', 'DomainName', 'Email']:
+        if key not in data:
+           #log.error('Key: "{}" is not set'.format(key))
+            print(json.dumps({'message': 'Key is not set', 'key': key}, default=str))
+            sys.exit('Key: "{}" is not set'.format(key))
```

Comment:
Please don't use less readable outdated constructions like `.format`. Use f-stirings which exist from python 3.6
```suggestion
            sys.exit(f'Key: "{key}" is not set')
```

Reply:
I will check it later.
I is just matter of taste, you are allowed to change it, but you should test it then.